test/MVCFramework45.FunctionalTests/Controllers/HomeController.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.ApplicationInsights.DataContracts;
+using Microsoft.ApplicationInsights;
+
+namespace MVCFramework45.FunctionalTests.Controllers
+{
+    public class HomeController : Controller
+    {
+        private TelemetryClient telemetryClient;
+
+        public HomeController(TelemetryClient telemetryClient)
+        {
+            this.telemetryClient = telemetryClient;
+        }
+
+        //private IApplicationEnvironment applicationEnvironment;
+
+        //public HomeController(TelemetryClient telemetryClient, IApplicationEnvironment applicationEnvironment)
+        //{
+        //    this.telemetryClient = telemetryClient;
+        //    this.applicationEnvironment = applicationEnvironment;
+        //}
+
+        public IActionResult Index()
+        {
+            return View();
+        }
+
+        public IActionResult Exception()
+        {
+            throw new InvalidOperationException("Do not call the method called Exception");
+        }
+
+        public IActionResult About(int index)
+        {
+            ViewBag.Message = "Your application description page # " + index;
+
+            return View();
+        }
+
+        public async Task<IActionResult> Contact()
+        {
+            this.telemetryClient.TrackEvent("GetContact");
+            this.telemetryClient.TrackMetric("ContactFile", 1);
+            this.telemetryClient.TrackTrace("Fetched contact details.", SeverityLevel.Information);
+            return View();
+        }
+
+        public IActionResult Error()
+        {
+            return View("~/Views/Shared/Error.cshtml");
+        }
+
+        //private async Task<string> GetContactDetails()
+        //{
```

Comment:
That's not the right namespace for that interface. You want this: [https://github.com/aspnet/PlatformAbstractions/blob/9668a6853bc663ed32733f068a06d10153e095cd/src/Microsoft.Extensions.PlatformAbstractions/IApplicationEnvironment.cs](https://github.com/aspnet/PlatformAbstractions/blob/9668a6853bc663ed32733f068a06d10153e095cd/src/Microsoft.Extensions.PlatformAbstractions/IApplicationEnvironment.cs)

However, note that this interface will go away in RTM. @pranavkm is working on a change for that, which once it's done you will see posted in [https://github.com/aspnet/Announcements/issues/](https://github.com/aspnet/Announcements/issues/)

---
In reply to: [60813732](https://github.com/Microsoft/ApplicationInsights-aspnetcore/pull/164#discussion_r60813732) [](ancestors = 60813732)